src/solvers/refinement/string_refinement.cpp

```diff
@@ -1466,16 +1466,18 @@ class find_index_visitort: public const_expr_visitort
 {
 private:
   const exprt &str_;
+  const symbol_exprt &qvar_;
 
 public:
-  explicit find_index_visitort(const exprt &str): str_(str) {}
+  explicit find_index_visitort(
+    const exprt &str, const symbol_exprt &qvar): str_(str), qvar_(qvar) {}
 
   void operator()(const exprt &expr)
   {
     if(expr.id()==ID_index)
     {
       const index_exprt &i=to_index_expr(expr);
-      if(i.array()==str_)
+      if(i.array()==str_ && find_qvar(i.index(), qvar_))
         throw i.index();
```

Comment:
I agree with @romainbrenguier , I'd prefer this to be the return value of `operator()` instead of being `void` and throwing